{rnicolescu} ruff formatting

roxanan1996 opened the pull request:
Add support for linting tools and dependencies management.
The file needed are:

- pyproject.toml. This is used for managing dependencies and configuration
into a single file. At the moment, it is not written to be able to
install this project as a proper package, but it can be changed in the future.

- .pre-commit-config.yaml. This will run Ruff's linter and
formatter via pre-commit.
Initially all py files were excluded. Following commits removed them from the exclude
rule one by one. 

Local usage:
- to install packages run:
```
python -m pip install -e ".[dev]"
```

You can also use a virtual environment.
- Install git hook for pre-commit: 
```
pre-commit install
```

It also added github action to check pre-commit, in case developers
do not have the pre commit hook installed as recommended.

To setup Ruff with your current editor of choice, check out the
official doc https://docs.astral.sh/ruff/editors/setup/

For vim, I personally use 
```
Plug 'dense-analysis/ale'
```
...

```
" Disable linter for C file (gets rid of that annoying include error)
let g:ale_linters = {
\  'rust': ['analyzer'],
\ 'python': ['ruff'],
\}

let g:ale_fixers = { 'rust': ['rustfmt', 'trim_whitespace', 'remove_trailing_lines'],
            \ 'python': ['ruff', 'ruff_format'],
            \ '*': ['remove_trailing_lines', 'trim_whitespace'],
            \}

" Format when the file is saved (:w)
let g:ale_python_ruff_options = '--extend-select I'

" Ruff
let g:vimruff_default = "both"
let g:ale_completion_enabled = 1
let g:ale_fix_on_save = 1
let g:ale_lint_on_text_changed = 'never'
let g:ale_lint_on_enter = 0
```

roxanan1996 committed:
packaging: setup tooling framework

Add support for linting tools and dependencies management.
The file needed are:

- pyproject.toml. This is used for managing dependencies and configuration
into a single file. At the moment, it is not written to be able to
install this project as a proper package, but it can be changed in the future.

- .pre-commit-config.yaml. This will run Ruff's linter and
formatter via pre-commit.
At the moment, all *py are excluded from pre-commit formatting.

Local usage:
- to install packages run: python -m pip install -e ".[dev]"
You can also use a virtual environment.
- Install git hook for pre-commit: pre-commit install

Signed-off-by: Roxana Nicolescu <rnicolescu@ciq.com>

roxanan1996 committed:
Add .gitignore

Signed-off-by: Roxana Nicolescu <rnicolescu@ciq.com>

roxanan1996 committed:
check_kernel_commits.py: Ruff formatting

This was done automatically by ruff format. Default configuration is used.

Variable l was replaced by index due to #E741. It is not recommened
to use variables named 'I', 'O' or 'l' because they are hard to read.

Signed-off-by: Roxana Nicolescu <rnicolescu@ciq.com>

roxanan1996 committed:
ciq-cherry-pick.py: Ruff formatting

This was done automatically by ruff format. Default configuration is used.

Signed-off-by: Roxana Nicolescu <rnicolescu@ciq.com>

roxanan1996 committed:
ciq_helper.py: Ruff formatting

This was done automatically by ruff format. Default configuration is used.

Signed-off-by: Roxana Nicolescu <rnicolescu@ciq.com>

roxanan1996 committed:
github actions: Add pre-commit job

Runs pre-commit checks within Github Actions.
It uses the configuration file .pre-commit-config.yaml, defined recently.

Signed-off-by: Roxana Nicolescu <rnicolescu@ciq.com>

roxanan1996 committed:
jira_pr_check.py: Ruff formatting

This was done automatically by ruff format. Default configuration is used.

Note: Strings longer than 120 are kept for now.

Signed-off-by: Roxana Nicolescu <rnicolescu@ciq.com>

roxanan1996 committed:
release_config.py: Ruff formatting

This was done automatically by ruff format. Default configuration is used.

Signed-off-by: Roxana Nicolescu <rnicolescu@ciq.com>

roxanan1996 committed:
rolling-release-update.py: Ruff formatting

This was done automatically by ruff format. Default configuration is used.

Important:
is_rebuild variable was never used. Therefore I removed this:
```
if b"Rebuild rocky" in line:
	is_rebuild = True
```

Signed-off-by: Roxana Nicolescu <rnicolescu@ciq.com>

roxanan1996 commented:
All *py are formatted now.

Notes:

1. Strings longer than 120 is ignored for now. Ruff does not format that automatically,
that's a downside.
I hope with some refactoring, we'll actually don't need such strong strings, because the
current scripts use many level of identation.

2. the github action for pre-commit checks use a user defined action.
Not sure what's the policy on this. I checked the code and it looks legit.
If required, I can just add my own rule based on it.

bmastbergen reviewed (commented):
The changes look fine to me.  We have typically shied away from using third party github actions.  Not sure if we are ok with that here or not.

PlaidCat commented:
> The changes look fine to me. We have typically shied away from using third party github actions. Not sure if we are ok with that here or not.

I thought I commented on this, Yes we want to avoid non-official, or not maintained by a major company. With the Supply Chain Attacks that have been going on for the past couple years and we don't have a great way of maintaining an audit 

Ok So I'm the one is backed by astral https://astral.sh/
The individuals, and no offense to them, its just a user pushing things: https://github.com/GriceTurrble/precommit-checks-action with no other contributes or reviewers.  Lets find or develop an alternative to the intention here.

roxanan1996 commented:
> > The changes look fine to me. We have typically shied away from using third party github actions. Not sure if we are ok with that here or not.
> 
> I thought I commented on this, Yes we want to avoid non-official, or not maintained by a major company. With the Supply Chain Attacks that have been going on for the past couple years and we don't have a great way of maintaining an audit
> 
> Ok So I'm the one is backed by astral https://astral.sh/ The individuals, and no offense to them, its just a user pushing things: https://github.com/GriceTurrble/precommit-checks-action with no other contributes or reviewers. Lets find or develop an alternative to the intention here.

Ok. I'll push our own asap.

roxanan1996 committed:
run_interdiff.py: Ruff formatting

This was done automatically by ruff format. Default configuration is used.

Hence this was the last file ignored, it removes the exclude rule in
.pre-commit-config.yaml

Signed-off-by: Roxana Nicolescu <rnicolescu@ciq.com>

roxanan1996 commented:
Ok, I removed the user defined github action. And it seems to be working fine.

bmastbergen reviewed (approved):
:curling_stone:

PlaidCat reviewed (approved):
:shipit:
